datatable/java/datatable/src/main/java/io/cucumber/datatable/DataTableTypeRegistry.java

```diff
@@ -26,69 +26,79 @@
-
-        TableCellTransformer<Integer> integerTableCellTransformer = applyIfPresent(Integer::decode);
-        defineDataTableType(new DataTableType(Integer.class, integerTableCellTransformer));
-        defineDataTableType(new DataTableType(int.class, integerTableCellTransformer));
-
-        TableCellTransformer<Long> longTableCellTransformer = applyIfPresent(Long::decode);
-        defineDataTableType(new DataTableType(Long.class, longTableCellTransformer));
-        defineDataTableType(new DataTableType(long.class, longTableCellTransformer));
-
-        TableCellTransformer<Float> floatTableCellTransformer = applyIfPresent(numberParser::parseFloat);
-        defineDataTableType(new DataTableType(Float.class, floatTableCellTransformer));
-        defineDataTableType(new DataTableType(float.class, floatTableCellTransformer));
-
-        TableCellTransformer<Double> doubleTableCellTransformer = applyIfPresent(numberParser::parseDouble);
-        defineDataTableType(new DataTableType(Double.class, doubleTableCellTransformer));
-        defineDataTableType(new DataTableType(double.class, doubleTableCellTransformer));
-
-        defineDataTableType(new DataTableType(Optional.class, transformAsOptional(objectTableCellTransformer)));
-
-        Type optionalString = new TypeReference<Optional<String>>() { }.getType();
-        defineDataTableType(new DataTableType(optionalString, transformAsOptional(objectTableCellTransformer)));
-
-        Type optionalDouble = new TypeReference<Optional<Double>>() { }.getType();
-        defineDataTableType(new DataTableType(optionalDouble, transformAsOptional(doubleTableCellTransformer)));
-
-        Type optionalFloat = new TypeReference<Optional<Float>>() { }.getType();
-        defineDataTableType(new DataTableType(optionalFloat, transformAsOptional(floatTableCellTransformer)));
-
-        Type optionalLong = new TypeReference<Optional<Long>>() { }.getType();
-        defineDataTableType(new DataTableType(optionalLong, transformAsOptional(longTableCellTransformer)));
-
-        Type optionalByte = new TypeReference<Optional<Byte>>() { }.getType();
-        defineDataTableType(new DataTableType(optionalByte, transformAsOptional(byteTableCellTransformer)));
-
-        Type optionalBigDecimal = new TypeReference<Optional<BigDecimal>>() { }.getType();
-        defineDataTableType(new DataTableType(optionalBigDecimal, transformAsOptional(bigDecimalTableCellTransformer)));
-
-        Type optionalBigInteger = new TypeReference<Optional<BigInteger>>() { }.getType();
-        defineDataTableType(new DataTableType(optionalBigInteger, transformAsOptional(bigIntegerTableCellTransformer)));
+        {
+            TableCellTransformer<Object> objectTableCellTransformer = applyIfPresent(s -> s);
+            defineDataTableType(new DataTableType(Object.class, objectTableCellTransformer, true));
+            defineDataTableType(new DataTableType(Optional.class, optionalIfAbsent(objectTableCellTransformer), true));
```

Comment:
`true` here marks the `DataTableType` as replaceable. This means that it's possible to register a new transformer that replaces this one. However this means that the optional transformer is not replaced. This seems somewhat inconsistent.

I would expect that when I replace the `objectTableCellTransformer` for `Object`, the `Optional<Objec>t` transformer is used.  And with that in mind would also expect that by registering a transformer for `X` I would have also registered one for `Optional<X>`.

Reply:
So looks like we have to remove Optional transformers from constructor, and just define Optional<X> for any type in io.cucumber.datatable.DataTableTypeRegistry#defineDataTableType
Will that work?

Reply:
I don't know yet. This solution has the advantage of being conceptually consistent.

My suggestion isn't.